quint/src/ir/IRVisitor.ts

```diff
@@ -444,6 +444,9 @@ export function walkExpression(visitor: IRVisitor, expr: ir.QuintEx): void {
       break
     }
     case 'lambda':
+      if (visitor.definitionDepth !== undefined) {
```

Comment:
Why do we allow this to be undefined?

Reply:
Because some visitors don't need it, therefore it is an optional attribute. If it was a mandatory attribute, those visitors would have to declare and initialize it.